ui/ui-components/components/record/SampleRecordCard.tsx

```diff
@@ -0,0 +1,231 @@
+import Link from "next/link";
+import { useEffect, useMemo, useState } from "react";
+import { Button, Card, Col, Row, Table } from "react-bootstrap";
+import useSWR from "swr";
+import { useGrouparooModelContext } from "../../contexts/grouparooModel";
+import { ApiHook, UseApi } from "../../hooks/useApi";
+import { Actions, Models } from "../../utils/apiData";
+import { successHandler } from "../../utils/eventHandlers";
+import { grouparooUiEdition } from "../../utils/uiEdition";
+import StateBadge from "../badges/StateBadge";
+import ManagedCard from "../lib/ManagedCard";
+import SeparatedItems from "../lib/SeparatedItems";
+import LinkButton from "../LinkButton";
+import LoadingButton from "../LoadingButton";
+import AddSampleRecordModal from "./AddSampleRecordModal";
+
+interface Props {
+  execApi: ApiHook["execApi"];
+  properties: Models.PropertyType[];
+}
+
+interface RecordRow {
+  propertyKey: string;
+  propertyId: string;
+  propertyState: Models.PropertyType["state"];
+  value: string | number | boolean;
+}
+
+const isConfigUI = grouparooUiEdition() === "config";
+
+const SampleRecordCard: React.FC<Props> = ({ properties, execApi }) => {
+  const model = useGrouparooModelContext();
+
+  const [importing, setImporting] = useState(false);
+  const [addingRecord, setAddingRecord] = useState(false);
+  const [recordId, setRecordId] = useState(() => {
+    if (typeof window !== "undefined") {
+      return window.localStorage.getItem(`sampleRecord:${model.id}`);
+    }
+    return undefined;
+  });
+
+  const { data: record, mutate: mutateRecord } = useSWR(
```

Comment:
Let's talk more about SWR! I don't think I see implicitly what it gains us here.  Our `Client` already has some caching mechanisms built in.  perhaps we could replace those with SWR?  The rough API was that all the components could call GET whenever they wanted, but the client would be smart and returned cached data if it already had it. 

The cache area in the Client is here https://github.com/grouparoo/grouparoo/blob/main/ui/ui-components/client/client.ts#L180-L190

Reply:
I didn't realize that the client had a caching mechanism. I will roll back the changes for now but I can see how SWR could simplify the client while at the same time providing some extra niceties. But if we were to adopt it we should also revisit the client.